File: cpp/array/easy/26.删除有序数组中的重复项.cpp
/*
 * @lc app=leetcode.cn id=26 lang=cpp
 *
 * [26] 删除有序数组中的重复项
 */

// @lc code=start
class Solution
{
public:
    int removeDuplicates(vector<int> &nums)
    {
        if (nums.empty())
            return 0;

        int start = 0;
        for (int i = 1; i < nums.size(); i++) {
            if (nums[start] == nums[i]) {
                continue;
            } else {
                nums[start + 1] = nums[i];
                start++;
            }
        }

        return start + 1;
    }
};
// @lc code=end

File: golang/array/easy/26.删除有序数组中的重复项.go
/*
 * @lc app=leetcode.cn id=26 lang=golang
 * @lcpr version=30202
 *
 * [26] 删除有序数组中的重复项
 */

// @lc code=start
func removeDuplicates(nums []int) int {
	slow, fast := 0, 0
	for fast < len(nums) {
		if nums[slow] != nums[fast] {
			slow++
			nums[slow] = nums[fast]
		}

		fast++
	}

	return slow + 1
}

// @lc code=end

/*
// @lcpr case=start
// [1,1,2]\n
// @lcpr case=end

// @lcpr case=start
// [0,0,1,1,1,2,2,3,3,4]\n
// @lcpr case=end

*/

